[aotd] Log torch._functorch.config in tlparse

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147883


Adding torch._functorch.config to tlparse for better debugability.
E.g. https://github.com/pytorch/pytorch/pull/147638 happened only with `torch._functorch.config.view_replay_for_aliased_outputs=False` which is True by defautl


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

IvanKobzarev committed:
[aotd] Log torch._functorch.config in tlparse

[ghstack-poisoned]

bdhirsh commented on torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py:
```diff
@@ -485,6 +485,14 @@ def aot_dispatch_autograd(
                 inner_meta.bw_donated_idxs = fw_metadata.bw_donated_idxs
 
         if aot_config.enable_log:
+            trace_structured(
```
I think this is only going to run in the training path, but not the inference path? You might need the same treatment in `aot_dispatch_base`

bdhirsh reviewed (approved):
sgtm!

IvanKobzarev committed:
Update on "[aotd] Log torch._functorch.config in tlparse"



Adding torch._functorch.config to tlparse for better debugability.
E.g. https://github.com/pytorch/pytorch/pull/147638 happened only with `torch._functorch.config.view_replay_for_aliased_outputs=False` which is True by defautl


[ghstack-poisoned]

IvanKobzarev committed:
Update on "[aotd] Log torch._functorch.config in tlparse"



Adding torch._functorch.config to tlparse for better debugability.
E.g. https://github.com/pytorch/pytorch/pull/147638 happened only with `torch._functorch.config.view_replay_for_aliased_outputs=False` which is True by defautl


[ghstack-poisoned]

IvanKobzarev committed:
Update on "[aotd] Log torch._functorch.config in tlparse"



Adding torch._functorch.config to tlparse for better debugability.
E.g. https://github.com/pytorch/pytorch/pull/147638 happened only with `torch._functorch.config.view_replay_for_aliased_outputs=False` which is True by defautl


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[aotd] Log torch._functorch.config in tlparse"



Adding torch._functorch.config to tlparse for better debugability.
E.g. https://github.com/pytorch/pytorch/pull/147638 happened only with `torch._functorch.config.view_replay_for_aliased_outputs=False` which is True by defautl


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

IvanKobzarev commented:
@pytorchbot merge
